Preprocessing for @file gcc option

timmy-quinn opened the pull request:
Context:
gcc has an option to extract flags and options from a file. 
gcc @"file_name"


The file looks something like this
-o "output" "src.c"

This PR is adding a preprocessing step before the build log is parsed. The preprocessing step extracts the commands from within an intermediary 

Issue: https://github.com/nickdiego/compiledb/issues/142

timmy-quinn committed:
Preprocessing inline files

timmy-quinn committed:
removed unnecessary import

timmy-quinn committed:
Merge pull request #1 from timmy-quinn/preprocessing

Preprocessing

nickdiego commented:
Would it be possible to add tests to exercise the new code path?

Thanks for contributing!

timmy-quinn commented:
Hi nick, sorry for the delay. I have added tests to exercise the code path. In writing the tests, I also found some edge cases which were not handled, and updated the code to handle those.
